util/metrics/import_metrics_files.rb

```diff
@@ -0,0 +1,165 @@
+  # @attr [integer] end_epoch Epoch seconds for the end of the collection window
+  # @attr [string] metrics_dir The "output directory" of the metrics collector puppet module
+  # @attr [string] output_dir The directory to place the tarfile in.
+  # @attr [string] tar_file_name Name of the tar file to create with the metrics
+  # @attr [boolean] verbose Verbose output
+  # @attr [integer] poll_interval The "collection_frequency" of the metrics collector puppet module
+  # @attr [integer] archive_interval The frequency of the *_metrics_dir cron job for the metrics
+  #   collector module.  This is how often it tars up the json files.
+  #
+  class ImportMetricsFiles
+    # TODO: make this optional?
+    JSON2GRAPHITE_PATH = "~/git/puppet-metrics-viewer/json2graphite.rb"
+
+    # Initialize class
+    #
+    # @author Bill Claytor
+    #
+    # @param [string] results_dir The GPLT results directory to process
+    # @param [string] prefix The prefix to use when building the server tag
+    #
+    # @return [void]
+    #
+    # @example
+    #   initialize(results_dir, prefix)
+    #
+    def initialize(results_dir, prefix)
+      @results_dir = results_dir
+      @prefix = prefix
+      @id = @results_dir.split("_").last
+    end
+
+    # The main entry point to the import_metrics_files script
+    #
+    # @author Bill Claytor
+    #
+    # @return [void]
+    #
+    # @example
+    #   import_metrics_files
+    #
+    def import_metrics_files
+      pmc_dir = "#{@results_dir}/puppet-metrics-collector"
+      raise "Directory not found: #{pmc_dir}" unless File.directory? pmc_dir
+
+      puts "Importing puppet_metrics_collector files:"
+      puts " results_dir: #{@results_dir}"
+      puts " prefix: #{@prefix}"
+      puts " id: #{@id}"
+
+      service_dirs = Dir.glob("#{pmc_dir}/*").select { |f| File.directory? f }
+
+      puts " service directories:"
+      puts service_dirs
+      puts
+
+      # each service directory
+      service_dirs.each do |service_dir|
+        puts "Checking directory: #{service_dir}"
+
+        # each host directory
+        host_dirs = Dir.glob("#{service_dir}/*").select { |f| File.directory? f }
+
+        puts " host directories: "
+        puts host_dirs
+        puts
+
+        host_dirs.each do |host_dir|
+          import_metrics_files_for_host_dir(host_dir)
+        end
+      end
+    end
+
+    # Calls the json2graphite.rb script for the specified host directory
+    # with a server tag using the following pattern: prefix_id_hostname
+    #
+    # @author Bill Claytor
+    #
+    # @param [string] host_dir The host directory to import
+    #
+    # @return [void]
+    #
+    # @example
+    #   import_metrics_files
+    #
+    def import_metrics_files_for_host_dir(host_dir)
+      hostname = File.basename host_dir
+      server_tag = "#{@prefix}_#{@id}_#{hostname}"
+      pattern = "'#{host_dir}/*.json'"
+      cmd = "ruby #{JSON2GRAPHITE_PATH} --pattern #{pattern}" \
```

Comment:
Which ruby is this going to use (system or defined ruby for env)?

Reply:
My assumption was that it would be run by a user on their test runner using the default ruby for that system. Is there an expectation that it would be run in a different environment where the specific ruby to use would need to be specified? If so, how should that be handled?

Reply:
i just run the script without "ruby" before it.  Which I think does the same as this does

Reply:
Updated to remove `ruby` from cmd.